app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt

```diff
@@ -67,52 +70,66 @@ class MeetingRequestService : FirebaseMessagingService() {
+        val location = Pair(latitudeString.toDouble(), longitudeString.toDouble())
+        val locationAndMessage = Pair(message, location)
         MeetingRequestManager.meetingRequestViewModel?.removeFromMeetingRequestSent(senderUid) {
           MeetingRequestManager.meetingRequestViewModel?.updateInboxOfMessages {}
         }
         if (accepted) {
-          showNotification(senderName + MSG_RESPONSE_ACCEPTED, "")
-          MeetingRequestManager.meetingRequestViewModel?.addPendingLocation(senderUid) {}
+          MeetingRequestManager.meetingRequestViewModel?.confirmMeetingLocation(
+              senderUid, locationAndMessage) {
+                Log.e("LOCATION CONFIRMATION", "failed to confirm the meeting location")
+              }
+          if (!isAppInForeground()) {
+            showNotification(senderName + MSG_RESPONSE_ACCEPTED, "")
+          }
         } else {
-          showNotification(senderName + MSG_RESPONSE_REJECTED, "")
+          if (!isAppInForeground()) {
+            showNotification(senderName + MSG_RESPONSE_REJECTED, "")
+          }
         }
-      }
-      "MEETING CONFIRMATION" -> {
-        val locationString = remoteMessage.data["location"] ?: "null"
-        val latitudeString = locationString.split(", ")[0]
-        val longitudeString = locationString.split(", ")[1]
-        MeetingRequestManager.meetingRequestViewModel?.confirmMeetingLocation(
-            senderUid, Pair(message, Pair(latitudeString.toDouble(), longitudeString.toDouble()))) {
-              Log.e("MeetingRequestService", "error when confirmMeetingLocation : ${it.message}")
-            }
-        showNotification(senderName + MSG_CONFIRMATION, MSG_CONFIRMATION_INFO)
+        MeetingRequestManager.meetingRequestViewModel?.stopMeetingRequestTimer(senderUid, this)
       }
       "MEETING CANCELLATION" -> {
+        val message = remoteMessage.data["message"] ?: "null"
         val stringReason =
             when (message) {
-              "distance" -> DISTANCE_REASON_CANCELLATION
+              "DISTANCE" -> DISTANCE_REASON_CANCELLATION
+              "TIME" -> TIME_REASON_CANCELLATION
               else -> DEFAULT_REASON_CANCELLATION
             }
-        showNotification("Cancelled meeting with $senderName", stringReason)
         MeetingRequestManager.meetingRequestViewModel?.removeFromMeetingRequestInbox(senderUid)
         MeetingRequestManager.meetingRequestViewModel?.removeFromMeetingRequestSent(senderUid) {}
+        MeetingRequestManager.meetingRequestViewModel?.stopMeetingRequestTimer(senderUid, this)
+        showNotification("Cancelled meeting with $senderName", stringReason)
       }
       "ENGAGEMENT NOTIFICATION" -> {
+        val message = remoteMessage.data["message"] ?: "null"
         // Only show engagement notifications if app is in background
         // Commented out for now as the locations don't update in the background so the feature
         // can't work until that is changed
```

Comment:
Do you still need to comment on the check to see if the app is in the foreground or not?

Reply:
Eeee no, this is a design decision. Since there is no way in the app to see if a meeting request is cancelled (like a badge or a popup...), i decided to make the notification trigger no matter what

Reply:
Okay fine!